dbt/include/trino/macros/materializations/snapshot.sql

```diff
@@ -32,3 +33,24 @@
             {%- endfor %})
 
 {% endmacro %}
+
+
+{% macro build_snapshot_staging_table(strategy, sql, target_relation) %}
+    {% set temp_relation = make_temp_relation(target_relation) %}
+
+    {% set select = snapshot_staging_table(strategy, sql, target_relation) %}
+
+    {# --Based on configuration, materialize temporary relation as a view or as a table-- #}
+    {%- set snapshot_tmp_relation_type = config.get('snapshot_tmp_relation_type', 'view') -%}
+    {% if snapshot_tmp_relation_type == 'view' %}
+        {% do run_query(create_view_as(temp_relation, select)) %}
+    {% elif snapshot_tmp_relation_type == 'table' %}
+        {% do run_query(create_table_as(True, temp_relation, select)) %}
+    {% else %}
+        {% do exceptions.raise_compiler_error(
+          "Invalid snapshot_tmp_relation_type, should be view or table, got " ~ snapshot_tmp_relation_type
+        ) %}
+    {% endif %}
+
+    {% do return(temp_relation) %}
```

Comment:
Given we can override this macro, we can simply always create a staging table that doesn't inherit the table properties of the main table and our problem is solved. Why not just do that and never create a view?

A view comes with other challenges as it is not static. Snapshots are used for capturing all changes on a table. Using a view puts us at risk of losing track of changes for example if the table schema is changed during the snapshotting, we could have changes getting merged into the snapshot table that don't contain the values of newly added columns after the creation of the snapshot view.

Reply:
I believe a user has a specific requirement of setting location where they would like to do it and has specific reason to do so. 

In general, Trino is not transactional, meaning incremental materialization would face the same issue not even with a temp view but also with a temp table.

Reply:
That's not correct, every materialisation on top of a table will be reflecting the state as reflected by the metadata in the beginning of the query. By using a view the columns will be static while the data is dynamic, if the snapshot uses a last modified timestamp, any values for added columns since creating the view won't be inserted in the snapshot table, next time they will be ignored since the max modified timestamp in the snapshot table will think it already processed those values.

Reply:
I also think that if user explicitly specify table location, it may not be the best idea to create temporary table in a completely different location. 

Maybe it would be a good solution to add similar behaviour as when not specifying location property, which is to create subfolders for each table in schema folder:
`schema_folder/table`
`schema_folder/table_tmp`

Specyfing location property creates table directly under specified location, and temp table is attempted to be created in the same location:
`user/specified/location`
`user/specified/location` _error_

Maybe it could add subfolder for each table, just like when not specifying location?
`user/specified/location/table`
`user/specified/location/table_tmp`

Maybe it could be implemented in Trino, and new property like `create_subfolder_per_table` could be added to control this behaviour? WDYT?